File: scala/bob/src/main/scala/Bob.scala
/**
  * Created by mz on 2/20 020.
  */
class Bob {
  def hey(s: String): String = {
    // define "yell at" as "any letter appear" and "all letters are uppercase"
    if (s.foldLeft(false)((a, b) => a || b.isLetter) && s.equals(s.toUpperCase))
      "Whoa, chill out!"
    // define "question" as "end with ?"
    else if (s.endsWith("?"))
      "Sure."
    // empty saying.
    else if (s.trim.isEmpty)
      "Fine. Be that way!"
    else
      "Whatever."
  }
}

File: go/bob/bob.go
// This is a "stub" file.  It's a little start on your solution.
// It's not a complete solution though; you have to write some code.

// Package bob should have a package comment that summarizes what it's about.
// https://golang.org/doc/effective_go.html#commentary
package bob

import (
	"strings"
	"unicode"
)

func hasLetter(s string) bool {
	for _, c := range s {
		if unicode.IsLetter(c) {
			return true
		}
	}
	return false
}

// Hey should have a comment documenting it.
func Hey(remark string) string {
	remark = strings.TrimSpace(remark)
	switch true {
	case len(remark) == 0:
		return "Fine. Be that way!"
	case hasLetter(remark) && strings.ToUpper(remark) == remark:
		if strings.HasSuffix(remark, "?") {
			return "Calm down, I know what I'm doing!"
		}
		return "Whoa, chill out!"
	case strings.HasSuffix(remark, "?"):
		return "Sure."
	default:
		return "Whatever."
	}
}
